clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -5833,6 +5839,15 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
         CmdArgs.push_back("-fno-sycl-esimd-build-host-code");
     }
 
+    if (IsSYCLCUDACompat) {
+      Args.addOptInFlag(CmdArgs, options::OPT_fsycl_cuda_compat,
+                        options::OPT_fno_sycl_cuda_compat);
```

Comment:
Use of `addOptInFlag()` is a little odd here given that `IsSYCLCUDACompat` is already known to be true.

Reply:
true, but I haven't find a simpler function to render the arg using `options::OPT_fsycl_cuda_compat`. The alternative is `CmdArgs .push_back("-fsycl-cuda-compat")`. Quite common but quite dislike this (just like for the default args). Happy to change though.

Reply:
Ok, yeah, I looked and I don't see a better way to handle this either. Ok, nothing to see here.